ndcube/ndcube.py

```diff
@@ -857,16 +879,15 @@ def __mul__(self, value):
                 cube_unit = u.Unit('') if self.unit is None else self.unit
                 value_unit = value.unit
                 value = value.to_value()
-                new_data = self.data * value
                 new_unit = cube_unit * value_unit
             else:
                 return NotImplemented
         else:
-            new_data = self.data * value
             new_unit = self.unit
-        new_cube = self._new_instance_from_op(new_data, new_unit)
-        if new_cube.uncertainty is not None:
-            new_cube.uncertainty.array *= value
+        new_data = self.data * value
+        new_uncertainty = (type(self.uncertainty)(self.uncertainty.array * value)
```

Comment:
Is it guaranteed that the uncertainty has the same units as `data`? For example, could `data` be in J and uncertainty be expressed in mJ?

Reply:
It is guaranteed as there is only one unit associated with the cube.  Of course the user could enter them differently but the interpretation of an NDCube is that they are the same by definition.